app/src/main/java/com/allan/boardbuddies/fragments/SharedFragment.java

```diff
@@ -1,31 +1,157 @@
 package com.allan.boardbuddies.fragments;
 
+import android.app.Activity;
+import android.content.Intent;
 import android.os.Bundle;
 
+import androidx.activity.result.ActivityResult;
+import androidx.activity.result.ActivityResultCallback;
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.activity.result.contract.ActivityResultContracts.StartActivityForResult;
+import androidx.annotation.Nullable;
 import androidx.fragment.app.Fragment;
+import androidx.recyclerview.widget.DividerItemDecoration;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
 
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
 
+import com.allan.boardbuddies.activities.CanvasActivity;
+import com.allan.boardbuddies.activities.EditActivity;
+import com.allan.boardbuddies.models.Note;
+import com.allan.boardbuddies.NoteAdapter;
 import com.allan.boardbuddies.R;
+import com.google.android.material.floatingactionbutton.FloatingActionButton;
 
-/**
- * A simple {@link Fragment} subclass.
- * Use the {@link SharedFragment#newInstance} factory method to
- * create an instance of this fragment.
- */
-public class SharedFragment extends Fragment {
+import org.json.JSONException;
+import org.json.JSONObject;
 
-    public SharedFragment() {
-        // Required empty public constructor
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+
+public class SharedFragment extends Fragment implements NoteAdapter.OnNoteListener {
```

Comment:
I'm not sure if I want to keep the actual notepad functionality... if I don't then the note fragment will be changed to canvas and shared equates to shared canvases. Otherwise if I keep the notepad, then this fragment can just be changed to canvas fragment or board fragment

Reply:
Got it, but "Shared" doesn't mean much, so you can call it "SharedCanvas" at least